src/coreComponents/mainInterface/ProblemManager.cpp

```diff
@@ -411,10 +411,30 @@ void ProblemManager::parseXMLDocument( xmlWrapper::xmlDocument const & xmlDocume
     // Open mesh levels
     MeshManager & meshManager = this->getGroup< MeshManager >( groupKeys.meshManager );
     meshManager.generateMeshLevels( domain );
-    ElementRegionManager & elementManager = domain.getMeshBody( 0 ).getMeshLevel( 0 ).getElemManager();
-    topLevelNode = xmlProblemNode.child( elementManager.getName().c_str());
-    elementManager.processInputFileRecursive( topLevelNode );
 
+    domain.getMeshBodies().forSubGroups<MeshBody>([&]( MeshBody & meshBody )
+    {
+      string const meshBodyName = meshBody.getName();
+      GEOSX_LOG_RANK_0("MeshBody "<<meshBodyName );
+      ElementRegionManager & elementManager = meshBody.getMeshLevel( 0 ).getElemManager();
+
+      xmlWrapper::xmlNode elementRegionsNode = xmlProblemNode.child( elementManager.getName().c_str());
+      for( xmlWrapper::xmlNode regionNode : elementRegionsNode.children() )
+      {
+        string const regionName = regionNode.attribute( "name" ).value();
+        string const regionMeshBodyName = regionNode.attribute( "meshBody" ).value();
+        string const cellBlocks = regionNode.attribute( "cellBlocks" ).value();
+
+        if( regionMeshBodyName==meshBodyName )
```

Comment:
Element regions will be assigned to a mesh body iff this is true. For users using a single mesh body this may feel unnecessary when configuring their input files - perhaps when there's a single mesh body, all element regions are added to it by default?